AppCenterIdentity/AppCenterIdentity/MSIdentity.m

```diff
@@ -131,26 +142,136 @@ + (void)handleUrlResponse:(NSURL *)url {
+                                  completionBlock:^(MSALResult *result, NSError *e) {
+                                    // TODO: Implement error handling.
+                                    if (e) {
+                                    } else {
+                                      NSString __unused *accountIdentifier = result.account.homeAccountId.identifier;
+                                      self.accessToken = result.accessToken;
+                                    }
+                                  }];
+  }
+}
+
+#pragma mark - Private methods
+
+- (NSString *)identityConfigFilePath {
+  return [NSString stringWithFormat:@"%@/%@", kMSIdentityPathComponent, kMSIdentityConfigFilename];
+}
+
+- (BOOL)loadConfigurationFromCache {
+  NSData *configData = [MSUtility loadDataForPathComponent:[self identityConfigFilePath]];
+  self.identityConfig = [self deserializeData:configData];
+  if (self.identityConfig == nil || ![self.identityConfig isValid]) {
+    [self clearConfigurationCache];
+    self.identityConfig = nil;
+    return NO;
+  }
+  return YES;
+}
+
+- (void)downloadConfigurationWithETag:(nullable NSString *)eTag {
+
+  // Download configuration.
+  MSIdentityConfigIngestion *ingestion =
+      [[MSIdentityConfigIngestion alloc] initWithBaseUrl:@"https://mobilecentersdkdev.blob.core.windows.net" appSecret:self.appSecret];
+  [ingestion sendAsync:nil
+                   eTag:eTag
+      completionHandler:^(__unused NSString *callId, NSHTTPURLResponse *response, NSData *data, __unused NSError *error) {
+        MSIdentityConfig *config = nil;
+        if (response.statusCode == MSHTTPCodesNo304NotModified) {
+          MSLogInfo([MSIdentity logTag], @"Identity configuration hasn't changed.");
+        } else if (response.statusCode == MSHTTPCodesNo200OK) {
+          config = [self deserializeData:data];
+          if ([config isValid]) {
+            NSURL *configUrl = [MSUtility createFileAtPathComponent:[self identityConfigFilePath]
+                                                           withData:data
+                                                         atomically:YES
+                                                     forceOverwrite:YES];
+
+            // Store eTag only when the configuration file is created successfully.
+            if (configUrl) {
+
+              // Response header keys are case-insensitive but NSHTTPURLResponse contains case sensitive keys in Dictionary.
+              for (NSString *key in response.allHeaderFields.allKeys) {
+                if ([[key lowercaseString] isEqualToString:kMSETagResponseHeader]) {
+                  [MS_USER_DEFAULTS setObject:(_Nonnull id)response.allHeaderFields[key] forKey:kMSIdentityETagKey];
+                  break;
+                }
+              }
+            } else {
+              MSLogWarning([MSIdentity logTag], @"Couldn't create Identity config file.");
```

Comment:
I think it makes more sense to log an error instead of warning when we failed to create identity config file as the login flow won't work when this happens.

Reply:
This is not blocking anything. We already read the data but couldn't create a file. It will be retried  later. So, `warning` should be enough.